src/prefect/infrastructure/kubernetes.py

```diff
@@ -311,6 +329,36 @@ def get_client(self) -> Generator["CoreV1Api", None, None]:
             finally:
                 client.rest_client.pool_manager.clear()
 
+    def _get_infrastructure_pid(self, job_name: str) -> str:
+        """Generates a kubernetes pid string in the form of `<cluster_name>:<job_name>`."""
+        cluster_name = self._get_cluster_name()
+        job_pid = f"{cluster_name}:{job_name}"
+        return job_pid
+
+    def _parse_infrastructure_pid(self, infrastructure_pid: str) -> Tuple[str, str]:
+        """Splits a kubernetes infrastructure pid into its component parts."""
+        cluster_name, job_name = infrastructure_pid.rsplit(":", 1)
+        return cluster_name, job_name
+
+    def _get_cluster_name(self) -> str:
+        _, active_context = kubernetes.config.list_kube_config_contexts()
+        cluster_name = active_context["context"]["cluster"]
+        return cluster_name
+
+    def _set_config(self) -> None:
+        """Set the correct kubernetes configuration."""
```

Comment:
Let's expand on this docstring as mentioned above to cover threadsafe and add a TODO to investigate returning a client.